packages/react-components/react-jsx-runtime/src/jsx.ts

```diff
@@ -0,0 +1,71 @@
+import * as React from 'react';
+import {
+  SlotRenderFunction,
+  UnknownSlotProps,
+  slotInternalChildrenSymbol,
+  slotExternalChildrenSymbol,
+} from '@fluentui/react-utilities';
+
+/**
+ * Equivalent to React.createElement but supporting v9 slot API
+ */
+export function jsx<P extends {}>(
+  type: React.ElementType<P>,
+  props?: P | null,
+  ...children: React.ReactNode[]
+): React.ReactElement<P> | null {
+  return isSlotComponent(type)
+    ? jsxFromSlotComponent(type, props, ...children)
+    : React.createElement(type, props, ...children);
+}
+
+function jsxFromSlotComponent<Props extends UnknownSlotProps>(
+  component: React.ElementType<Props>,
+  props?: Props | null,
+  ...overrideChildren: React.ReactNode[]
+): React.ReactElement<Props> | null {
+  const children = normalizeChildren(props, overrideChildren);
+  const {
+    [slotInternalChildrenSymbol as keyof Props]: _,
+    [slotExternalChildrenSymbol as keyof Props]: externalChildren,
+    ...propsWithoutSymbols
+  } = props ?? {};
+
+  if (typeof externalChildren === 'function') {
+    const render = externalChildren as SlotRenderFunction<Props>;
+    return React.createElement(
+      React.Fragment,
+      {},
+      render(component, { ...propsWithoutSymbols, children } as Omit<Props, 'as'>),
+    ) as React.ReactElement<Props>;
+  }
+
+  return React.createElement<Props>(component, { ...propsWithoutSymbols, children } as Props);
+}
+
+export function extractChildrenFromProps<Props extends React.PropsWithChildren<{}>>(
+  props?: Props | null,
+): React.ReactNode[] {
+  if (!props) {
+    return [];
+  }
+  return Array.isArray(props.children) ? props.children : [props.children];
+}
+
+function normalizeChildren<Props extends UnknownSlotProps>(
+  props?: Props | null,
+  overrideChildren: React.ReactNode[] = [],
+): React.ReactNode | undefined {
+  const internalChildren = props && (props[slotInternalChildrenSymbol as keyof Props] as React.ReactNode);
+  const externalChildren =
+    props && (props[slotExternalChildrenSymbol as keyof Props] as React.ReactNode | SlotRenderFunction<Props>);
```

Comment:
```suggestion
function normalizeChildren<Props extends UnknownSlotProps>(
  props: Props,
  overrideChildren: React.ReactNode[] = [],
): React.ReactNode | undefined {
  const internalChildren = props[slotInternalChildrenSymbol as keyof Props] as React.ReactNode;
  const externalChildren = props[slotExternalChildrenSymbol as keyof Props] as
    | React.ReactNode
    | SlotRenderFunction<Props>;

```

Due `isSlotComponent()` check, `props` can be `null` or `undefined` at this point, correct?

Reply:
Yeah, I can update `isSlotComponent` signature to ensure `props` is non nullable, it'll simplify things